File: Assets/MLRepo/ml.72201f1e-ba18-487e-adc4-4381ae10e21b.davidmatos_mirroroptimizationvolume/MirrorOptimizationVolume/Scripts/MirrorSwitch.cs
using ML.SDK;
using UnityEngine;
    public class MirrorSwitch : MonoBehaviour
    {
        //|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
        [Header("Mirror References")]
        [SerializeField] private GameObject activeMirror; // High-quality mirror to activate when player is near
        [SerializeField] private GameObject fallbackMirror; // Low-quality mirror to use when player is far

        //|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
        private MLMirror mirrorComponent; // Reference to the MLMirror component on the active mirror

        //|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
        private void Start()
        {
            if (activeMirror != null)
            {
                mirrorComponent = activeMirror.GetComponent(typeof(MLMirror)) as MLMirror;
                if (mirrorComponent != null)
                {
                    mirrorComponent.MirrorActive = false;
                }
                activeMirror.SetActive(false);
            }

            if (fallbackMirror != null)
            {
                fallbackMirror.SetActive(true);
            }
        }

        private void Update()
        {
            // Not used (kept for consistency with original)
        }

        // NOTE: The local player is the only one that will trigger these
        private void OnTriggerEnter(Collider other)
        {
            // Using non-generic GetComponent for MLPlayer script
            MLPlayer player = other.gameObject.GetPlayer();
            if (player != null)
            {
                if (activeMirror != null)
                {
                    activeMirror.SetActive(true);
                }

                if (fallbackMirror != null)
                {
                    fallbackMirror.SetActive(false);
                }

                if (mirrorComponent != null)
                {
                    mirrorComponent.MirrorActive = true;
                }
            }
        }

        // NOTE: The local player is the only one that will trigger these
        private void OnTriggerExit(Collider other)
        {
            // Using non-generic GetComponent for MLPlayer script
            MLPlayer player = other.gameObject.GetPlayer();
            if (player != null)
            {
                if (mirrorComponent != null)
                {
                    mirrorComponent.MirrorActive = false;
                }

                if (activeMirror != null)
                {
                    activeMirror.SetActive(false);
                }

                if (fallbackMirror != null)
                {
                    fallbackMirror.SetActive(true);
                }
            }
        }
    }

File: Assets/MLRepo/ml.72201f1e-ba18-487e-adc4-4381ae10e21b.davidmatos_mirroroptimizationvolume/MirrorOptimizationVolume/Scripts/MirrorSwitch.lua
Namespace("davidmatos_mirroroptimizationvolume");
do -- script MirrorSwitch 
	
	-- get reference to the script
	local MirrorSwitch = LUA.script;
	
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--these variables are exposed and accessible through the unity inspector

	local gameObject_activeMirror = SerializedField("Active Mirror", GameObject); 
	local gameObject_fallbackMirror = SerializedField("Fallback Mirror", GameObject); 

	--|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||

	local mlmirror_mirror = nil;

	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||

	-- start only called at beginning
	function MirrorSwitch.Start()
		mlmirror_mirror = gameObject_activeMirror.GetComponent(MLMirror);

		mlmirror_mirror.MirrorActive = false;
		gameObject_activeMirror.SetActive(false);

		if(gameObject_fallbackMirror ~= nil) then
			gameObject_fallbackMirror.SetActive(true);
		end
	end

	-- update called every frame
	function MirrorSwitch.Update()

	end

	--NOTE TO SELF: The local player is the only one that seems to trip these
	function MirrorSwitch.OnTriggerEnter(other)
		--if the object that entered the trigger volume is infact a player
		if(other.gameObject.IsPlayer() == true) then
			gameObject_activeMirror.SetActive(true);

			if(gameObject_fallbackMirror ~= nil) then
				gameObject_fallbackMirror.SetActive(false);
			end

			mlmirror_mirror.MirrorActive = true;
		end
	end

	--NOTE TO SELF: The local player is the only one that seems to trip these
	function MirrorSwitch.OnTriggerExit(other)
		--if the object that entered the trigger volume is infact a player
		if(other.gameObject.IsPlayer() == true) then
			mlmirror_mirror.MirrorActive = false;
			gameObject_activeMirror.SetActive(false);

			if(gameObject_fallbackMirror ~= nil) then
				gameObject_fallbackMirror.SetActive(true);
			end
		end
	end
end